WIP: Added doc overview for devfiles

jc-berger opened the pull request:
For issue [#159](https://github.com/devfile/api/issues/159). 

Possible suggestions for more revisions:

1. An "additional resources" section where we can have a visual or audible demonstration of devfiles. For instance, I used this YouTube video to help me learn: https://www.youtube.com/watch?v=vnsLwtRz--w&ab_channel=EclipseFoundation 
2. A one sentence definition for `API Reference` and `Guide for Stack Authors` and all the others we include here, so users know what they'll be reading before clicking away. 

To any reviewer, let me know what you think, thanks!

johnmcollier commented on docs/modules/ROOT/pages/index.adoc:
```diff
@@ -1,9 +1,11 @@
 :description: Software Defined Development Environments
 
-Devfiles describe and define a cloud native development environment.
+Devfiles are YAML files that describe and define an environment in Che, including the source code and development components, like IDE tools and application runtimes. Che uses these devfiles to make a cloud workspace composed of multiple containers.
```
Devfiles can be used outside of Che in other tools,such as in Odo, and eventually the OpenShift console, among other tools. 

Personally I prefer the generic "cloud native development environment" since it's not specific to any one tool or platform.

jc-berger replied:
Thanks, John!

elsony replied:
+1 for using cloud native development environment and making the description generic and not restricting that to Che.

rkratky replied:
@jc-berger, hyphenate compound adjectives.
```suggestion
Devfiles are YAML files that describe and define cloud-native development environments, including source code and development components, like IDE tools and application runtimes. Various applications, including link:https://www.eclipse.org/che/[Eclipse Che], use devfiles to create cloud workspaces composed of multiple containers.
```

elsony commented on docs/modules/ROOT/pages/index.adoc:
```diff
@@ -23,3 +25,7 @@ See xref:migration_guide.adoc[Devfile v2 Migration Guide].
+
+== Additional Resources
+
+For a visual and audible demonstration of devfiles, see the YouTube video, https://youtu.be/vnsLwtRz--w[Customize your own online IDE with a devifle].
```
That video is based on devfile 1.0. Although some of the concepts that it shows still apply, the sample devfile that it uses don't match devfile 20 so I wonder if we should include that as part of this page. It may introduce confusion if the user tries to follow what shows on the video.

elsony reviewed (changes requested):
@jc-berger I think there is a lot more background info that we can provide on the overview page to better explain and provide an introduction to devfile. I have got a deck that contains some overview info on devfile. I'll send that to you so you can beef up the documentation based on that.

rkratky commented on docs/modules/ROOT/pages/index.adoc:
```diff
@@ -1,9 +1,11 @@
+Devfiles are YAML files that describe and define an environment in Che, including the source code and development components, like IDE tools and application runtimes. Che uses these devfiles to make a cloud workspace composed of multiple containers.
 
-Devfiles are consumed by tools that transform the definition into a cloud workspace composed of multiple cloud native resources. Any number of identical workspaces can be created from the same devfile. They can be shared in various ways including a devfile registry or together with the source code of a project.
+Within your devfile workspace, you can create multiple projects. In the components of the project, like the editor, plugin, or environment, the devfile commands build, test, and run your application. Also, any number of workspaces can be created from the same devfile. This way, workspaces can be shared and worked on by different teams. So now you can easily transfer, collaborate and expand your w ...
```
@jc-berger:

* Unless necessary, avoid using "you", "your", or other personal pronouns. There's no need for a devfile, projects, or workspaces to belong to the reader.
* Use "plug-in" (see _The IBM Style Guide_ pg. 353).
* Use the serial comma (see _The IBM Style Guide_ pg. 241).
* Avoid marketing speak, including "easily" or "now".
* Don't edit text just for the sake of editing it. In this case, the orig. first sentence was perfectly fine and better described the devfile concept.

rkratky commented on docs/modules/ROOT/pages/index.adoc:
```diff
@@ -23,3 +25,7 @@ See xref:migration_guide.adoc[Devfile v2 Migration Guide].
 
 See https://github.com/devfile/api/tree/master/samples/devfiles[Devfile Samples].
+
+== Additional Resources
```
Don't use title case in headings.
```suggestion
== Additional resources
```

jc-berger commented:
Due to the doc restructure, the doc I edited in this PR no longer exists., making a merge very difficult. Therefore, it was easier to make a new branch and PR. I've addressed all the comments and applied all the feedback in the new PR, #28. 

Closing this PR since we'll focus our attention on #28, from hereon out.
